pkg/reporting/cookbooks.go

```diff
@@ -158,7 +159,7 @@ type cookbookItem struct {
 func NewCookbooksReport(
 	cbi CookbookInterface, cbai CBAInterface, pgi PolicyGroupInterface, poi PolicyInterface, searcher SearchInterface,
 	runCookstyle bool, onlyUnused bool, workers int,
-	nodeFilter string,
+	nodeFilter string, anonymize bool,
```

Comment:
I'm starting to question the wisdom of a `New` that wants all of these as arguments. This is getting much more cumbersome than originally forseen. 
Not for this PR, but something to consider is moving all of these interface definitions into go-chef as part of the API client. The only reason we have to declare them individually here is that we didn't want to write a multi-layered interface & mock for `client`.

Reply:
Agreed that it is getting overly complex. I think main reason is the ease of testing. We can brainstorm what would be a good strategy to properly refactor.